integration_tests/test_ifexp_03.py

```diff
@@ -0,0 +1,11 @@
+from lpython import i32
+
+def fib(n: i32) -> i32:
+    return fib(n - 1) + fib(n - 2) if n >= 3 else 1
+
+def main0():
+    res: i32 = fib(30)
```

Comment:
@Shaikh-Ubaid BTW, just in case if you don't know, you can use less than fib(30) value (smaller). This is to make test run faster. Even for fib(10) there will be a crash of recursion overflow in older (unfixed) code.

Reply:
Thank you for sharing @polkovnikov! Yes, I agree `fib(10)` reproduces the error and could be faster that `fib(30)`. I used `fib(30)`, because:
- the test case with `fib(30)` seems to execute quickly on my system
```bash
$ time python integration_tests/test_ifexp_03.py
832040
python integration_tests/test_ifexp_03.py  0.13s user 0.01s system 98% cpu 0.147 total
$ time lpython integration_tests/test_ifexp_03.py
832040
lpython integration_tests/test_ifexp_03.py  0.10s user 0.02s system 102% cpu 0.117 total
```
- I think `fib(30)` might have much more recursion calls and therefore, huge stack space utilization due to it. Thus, I think the test also checks the working of the binary when there are many recursion calls and huge stack space utilization due to it. 


Also thank you so much for trying LPython and reporting the bug! We appreciate it.